web-animations-1/Overview.bs

```diff
@@ -833,14 +833,64 @@ follows:
     <var>animation</var>, if any.
 1.  If <var>new timeline</var> is the same object as <var>old timeline</var>,
     abort this procedure.
+1.  Let |previous play state| be |animation|'s [=play state=].
+1.  Let |previous current time| be the |animation|'s [=current time=].
+1.  Let |from finite timeline| be true if |old timeline| is not null and
+    not [=monotonically increasing=].
+1.  Let |to finite timeline| be true if |timeline| is not null and not
+    [=monotonically increasing=].
 1.  Let the <a>timeline</a> of <var>animation</var> be <var>new timeline</var>.
+1.  Set the flag |reset current time on resume| to false.
+1.  Perform the steps corresponding to the <em>first</em> matching
+    condition from the following, if any:
+
+    <div class="switch">
+
+    :   If |to finite timeline|,
+
+    ::  1.  [=Apply any pending playback rate=] on |animation|
+        1.  Let |seek time| be zero if [=playback=] rate &ge; 0, and
+            |animation|'s <a>associated effect end</a> otherwise.
+        1.  Update the animation based on the first matching condition if any:
+
+            <div class="switch">
+
+            :   If <em>either</em> of the following conditions are true:
+                *    |previous play state| is 'running' or,
+                *    |previous play state| is 'finished'
+
+            ::  Set |animation|'s [=start time=] to |seek time|.
+
+            :   If |previous play state| is 'paused':
+
+            ::  1.  Set the flag |reset current time on resume| to true.
+                1.  Set [=start time=] to unresolved.
+                1.  Set [=hold time=] to |previous current time|.
+
+                <p class="note">
+                This step ensures that the [=current time=] is preserved even in
+                the case of a pause-pending animation with a resolved
```

Comment:
If previous current time is unresolved and animation in pause-pending state, this treatment creates a new state which is not currently handled:
start_time=unresolved;
current_time=unresolved;
state = pause-pending;

How about setting start time to zero instead?

Reply:
Right, this state would arise if the previous timeline was inactive and previously in a pause-pending state with start time resolved and hold time unresolved.

Reply:
Good catch.  Updated the pull request.

Will also look into moving this pull request to V2 of the spec.  I believe we are backward compatible with the current procedure insofar as we do not introduce behavioral changes when transitioning between document timelines or between a document timeline and null timeline.